### PR TITLE
variable naming for Distilbert model

### LINKED ISSUE #16051: TF: clearer model variable naming
### This issue is part of our **Great Code Cleanup 2022**. If you're interested in helping out, take a look at [this thread](https://twitter.com/carrigmat/status/1502319813510766599), or come [join us on Discord](https://t.co/kS42XBvpWH) and talk with other contributors!

As introduced in https://github.com/huggingface/transformers/issues/15908 and implemented in https://github.com/huggingface/transformers/pull/15907, we now have a new `@unpack_inputs` decorator to unpack TensorFlow model `call()` arguments. In essence, if we apply the decorator, we can replace `inputs["foo"]` with `foo`, making the code for the layer/model much shorter and clearer.

This issue is a call for contributors, to implement the new decorator in the architectures below. If you wish to contribute, reply in this thread which architectures you'd like to take :)

### Guide to contributing:
1. Ensure you've read our contributing [guidelines](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md) 📜 
2. Claim your architecture(s) in this thread (confirm no one is working on it) 🎯 
3. Implement the changes as in https://github.com/huggingface/transformers/pull/15907 (see the diff on the model architectures for a few examples) 💪 
    - The file you want to look at is in `src/transformers/models/[model_name]/modeling_tf_[model_name].py`
    - In functions that have an `input_processing` call, remove it and add the `@unpack_inputs` decorator
    - Replace any use of the `inputs` variable in those functions (e.g. `inputs["foo"]` -> `foo`)
5. Test your changes on the architecture with `RUN_SLOW=1 py.test -vv tests/[model_name]/test_modeling_tf_[model_name].py` 💻 
6. Open the PR and tag me in it (don't forget to run `make fixup` before your final commit) 🎊 
    - Note that some code is copied across our codebase. If you see a line like `# Copied from transformers.models.bert...`, this means that the code is copied from that source, and our scripts will automatically keep that in sync. If you see that, you should not edit the copied method! Instead, edit the original method it's copied from, and run make fixup to synchronize that across all the copies. Be sure you installed the development dependencies with `pip install -e ".[dev]"`, as described in the contributor guidelines above, to ensure that the code quality tools in `make fixup` can run.

### Models updated:
- [x] (the templates)
- [x] albert
- [x] auto
- [x] bart
- [x] bert
- [x] blenderbot
- [x] blenderbot_small
- [x] camembert
- [x] clip
- [x] convbert
- [x] convnext
- [x] ctrl
- [x] deberta
- [x] deberta_v2
- [x] distilbert
- [x] dpr
- [x] electra
- [x] encoder_decoder
- [x] flaubert
- [x] funnel
- [x] gpt2
- [x] hubert
- [x] layoutlm
- [x] led
- [x] longformer
- [x] lxmert
- [x] marian
- [x] mbart
- [x] mobilebert
- [x] mpnet
- [x] mt5
- [x] openai
- [x] pegasus
- [x] rag
- [x] rembert
- [x] roberta
- [x] roformer
- [x] speech_to_text
- [x] t5
- [x] tapas
- [x] transfo_xl
- [x] vision_encoder_decoder
- [x] vit
- [x] wave2vec2 👉  lgnore this one for now, looking into the issue @infinite-Joy raised
- [x] xlm
- [x] xlm_roberta
- [x] xlnet

### PR DESCRIPTION
**Pull request by robotjellyzone**
# What does this PR do?
As part of #16051 this PR changes the unpacking of the inputs of TF Distilbert model to use the decorator instead.

See #16051

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@Rocketknight1 @gante

### CODE REVIEW

File: src/transformers/models/distilbert/modeling_tf_distilbert.py [hunk 1]
```diff
@@ -547,6 +534,7 @@ def __init__(self, config, *inputs, **kwargs):
         output_type=TFBaseModelOutput,
         config_class=_CONFIG_FOR_DOC,
     )
+    @unpack_inputs
```
Comment (gante) [inline src/transformers/models/distilbert/modeling_tf_distilbert.py:537]: In the other PRs, `@unpack_inputs` has been placed as the top decorator. Can you please move it so as to be the top decorator (here and in the other uses of `@unpack_inputs`) 🙏

**Review by gante — COMMENTED** [review]
Awesome, thank you for the contribution 🔥  In addition to the comment below, can you please:
1. Run `make fixup` before your next commit (which should fix the failing test)
2. Confirm that the tests pass locally (`RUN_SLOW=1 py.test -vv tests/[model_name]/test_modeling_tf_[model_name].py`)

File: src/transformers/models/distilbert/modeling_tf_distilbert.py (hunk 1 above)
Comment (Rocketknight1) [reply src/transformers/models/distilbert/modeling_tf_distilbert.py]: I'll run make fixup here, one sec!

Comment (robotjellyzone) [reply src/transformers/models/distilbert/modeling_tf_distilbert.py]: thank you so much @Rocketknight1 👍🏻

**Review by gante — APPROVED** [review]

**Comment by gante** [comment]
Nvm, reread your comment! Merging 👍

**Pull request merged**